src/components/markdown-renderer/replace-components/image/image-frame.tsx

```diff
@@ -18,11 +22,39 @@ export const ImageFrame: React.FC<React.ImgHTMLAttributes<HTMLImageElement>> = (
 
   if (imageProxyEnabled) {
     return (
-      <img alt={alt} src={imageUrl} title={title ?? ''} {...props}/>
+      <Frame src={imageUrl} title={title ?? ''} {...props}/>
     )
   }
 
   return (
-    <img alt={alt} src={src ?? ''} title={title ?? ''} {...props}/>
+    <Frame src={src ?? ''} title={title ?? ''} {...props}/>
+  )
+}
+
+const Frame: React.FC<React.ImgHTMLAttributes<HTMLImageElement>> = ({ alt, ...props }) => {
+  const { t } = useTranslation()
+  const [showFullscreenImage, setShowFullscreenImage] = useState(false)
+
+  return (
+    <Fragment>
+      <img alt={alt} {...props}/>
+      <div className={'text-right'} style={{ marginTop: '-31px' }}>
+        <Button size={'sm'} variant={'dark'} onClick={() => setShowFullscreenImage(true)} title={t('renderer.image.expandImage')}>
+          <ForkAwesomeIcon icon={'expand'}/>
+        </Button>
+      </div>
+      <CommonModal
+        show={showFullscreenImage}
+        onHide={() => setShowFullscreenImage(false)}
+        title={alt ?? ''}
+        closeButton={true}
+        size={'xl'}
+        icon={'picture-o'}
+      >
+        <Modal.Body>
+          <img alt={alt} {...props} className={'w-100'}/>
+        </Modal.Body>
+      </CommonModal>
```

Comment:
I don't see anything wrong with using bootstrap modals, because they are already included and provide most of the functionality. But I would remove the header, footer and padding.

Reply:
fixed